File: 2026/03- March/13-Friday/solution.py
import math
from typing import List

class Solution:
    def minNumberOfSeconds(self, mountainHeight: int, workerTimes: List[int]) -> int:
        
        def can_finish(time):
            total = 0
            
            for w in workerTimes:
                val = (2 * time) // w
                x = int((math.sqrt(1 + 4 * val) - 1) // 2)
                total += x
                
                if total >= mountainHeight:
                    return True
            
            return False
        
        left = 0
        right = min(workerTimes) * mountainHeight * (mountainHeight + 1) // 2
        
        while left < right:
            mid = (left + right) // 2
            
            if can_finish(mid):
                right = mid
            else:
                left = mid + 1
        
        return left

File: 2026/03- March/13-Friday/Solution.c
#include <math.h>
#include <stdbool.h>

bool canFinish(long long time, int mountainHeight, int* workerTimes, int workers) {
    long long total = 0;

    for (int i = 0; i < workers; i++) {
        long long w = workerTimes[i];
 
        long long val = (2 * time) / w;
        long long x = (long long)((sqrt(1.0 + 4.0 * val) - 1) / 2);

        total += x;

        if (total >= mountainHeight)
            return true;
    }

    return false;
}

long long minNumberOfSeconds(int mountainHeight, int* workerTimes, int workerTimesSize) {
    
    long long left = 0;
    long long right = (long long)workerTimes[0] * mountainHeight * (mountainHeight + 1) / 2;

    for (int i = 1; i < workerTimesSize; i++) {
        long long candidate = (long long)workerTimes[i] * mountainHeight * (mountainHeight + 1) / 2;
        if (candidate < right)
            right = candidate;
    }

    while (left < right) {
        long long mid = left + (right - left) / 2;

        if (canFinish(mid, mountainHeight, workerTimes, workerTimesSize))
            right = mid;
        else
            left = mid + 1;
    }

    return left;
}
